File: lab02a.py
#!/usr/bin/env python3
"""
Programming Assignment 2: Part One
"""


def longest_word(file_name):
    """
    Reads lines from a file and prints the length of the longest word
    for each line.
    """
    with open(file_name, encoding='utf-8') as file:
        for line in file:
            print(max(len(word) for word in line.split()))


longest_word('lab02a.txt')

File: lab02a.cpp
// Programming Assignment 2: Part One

#include <iostream>
#include <fstream>
#include <string>
#include <sstream>      // For std::stringstream
#include <cctype>      // For std::isspace
#include <limits>      // For std::numeric_limits

int main() {
    std::ifstream inputFile("lab02a.txt"); // Change to your input file
    if (!inputFile) {
        std::cerr << "Error opening file." << std::endl;
        return 1;
    }

    std::string line;
    while (std::getline(inputFile, line)) {
        std::istringstream lineStream(line);
        std::string word;
        int maxLength = 0;

        while (lineStream >> word) {
            if (word.length() > static_cast<size_t>(maxLength)) {
                maxLength = word.length();
            }
        }

        std::cout << maxLength << std::endl;
    }

    return 0;
}
